instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/server/OpenTelemetryService.java

```diff
@@ -52,20 +50,25 @@ private OpenTelemetryService(HttpService delegate, ArmeriaServerTracer serverTra
 
   @Override
   public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-    String route = route(ctx);
-    String spanName = route != null ? route : "HTTP " + req.method().name();
+    String spanName = ctx.config().route().patternString();
 
     // Always available in practice.
     long requestStartTimeMicros =
         ctx.log().ensureAvailable(RequestLogProperty.REQUEST_START_TIME).requestStartTimeMicros();
     long requestStartTimeNanos = TimeUnit.MICROSECONDS.toNanos(requestStartTimeMicros);
     Context context = serverTracer.startSpan(req, ctx, null, spanName, requestStartTimeNanos);
 
-    if (Span.fromContext(context).isRecording()) {
+    Span span = Span.fromContext(context);
+    if (span.isRecording()) {
       ctx.log()
           .whenComplete()
           .thenAccept(
               log -> {
+                if (log.responseHeaders().status() == HttpStatus.NOT_FOUND) {
```

Comment:
/cc @trustin @minwoox @ikhoon to confirm this is the most natural way of setting our route name to the path itself when not found.

Reply:
Though it may not matter - our unit test is checking that the span name is the path name, but for not found, shouldn't our instrumentation generally fallback to `HTTP {method}` @trask @iNikem?

Reply:
@anuraaga As you noticed, you can't distinguish who returned 404 (fallback or your service) with `log.responseHeaders().status() == HttpStatus.NOT_FOUND`.

If you did not bind any services on `Route.ofCatchAll()`, you could do this:

    ctx.config().route() == Route.ofCatchAll()

.. but this isn't a great experience. How about this:

    ctx.config().route().isFallback()

..? I guess we can add this in the next minor update.

Reply:
@trustin Yeah I think that could be helpful :)

Reply:
https://github.com/line/armeria/issues/3365